src/read_write/raw.rs

```diff
@@ -107,45 +109,228 @@ impl NodeReader for RawNodeReader {
+                (0..NUM_POINTS_PER_BATCH).try_for_each(|_| {
+                    let x = fixpoint_decode(
+                        self.xyz_reader.read_u16::<LittleEndian>()?,
+                        min.x,
+                        edge_length,
+                    );
+                    let y = fixpoint_decode(
+                        self.xyz_reader.read_u16::<LittleEndian>()?,
+                        min.y,
+                        edge_length,
+                    );
+                    let z = fixpoint_decode(
+                        self.xyz_reader.read_u16::<LittleEndian>()?,
+                        min.z,
+                        edge_length,
+                    );
+                    batch.position.push(Vector3::new(x, y, z));
+                    Ok(())
+                })
+            }
+            Encoding::ScaledToCube(min, edge_length, PositionEncoding::Float32) => {
+                (0..NUM_POINTS_PER_BATCH).try_for_each(|_| {
+                    let x = decode(
+                        self.xyz_reader.read_f32::<LittleEndian>()?,
+                        min.x,
+                        edge_length,
+                    );
+                    let y = decode(
+                        self.xyz_reader.read_f32::<LittleEndian>()?,
+                        min.y,
+                        edge_length,
+                    );
+                    let z = decode(
+                        self.xyz_reader.read_f32::<LittleEndian>()?,
+                        min.z,
+                        edge_length,
+                    );
+                    batch.position.push(Vector3::new(x, y, z));
+                    Ok(())
+                })
+            }
+            Encoding::ScaledToCube(min, edge_length, PositionEncoding::Float64) => {
+                (0..NUM_POINTS_PER_BATCH).try_for_each(|_| {
+                    let x = decode(
+                        self.xyz_reader.read_f64::<LittleEndian>()?,
+                        min.x,
+                        edge_length,
+                    );
+                    let y = decode(
+                        self.xyz_reader.read_f64::<LittleEndian>()?,
+                        min.y,
+                        edge_length,
+                    );
+                    let z = decode(
+                        self.xyz_reader.read_f64::<LittleEndian>()?,
+                        min.z,
+                        edge_length,
+                    );
+                    batch.position.push(Vector3::new(x, y, z));
+                    Ok(())
+                })
+            }
+        };
+
+        self.attribute_readers
+            .iter_mut()
+            .for_each(|(key, AttributeReader { dtype, reader })| {
+                match dtype {
+                    AttributeDataType::U8 => {
+                        let mut attr = Vec::with_capacity(NUM_POINTS_PER_BATCH);
+                        let _err: io::Result<()> = (0..NUM_POINTS_PER_BATCH).try_for_each(|_| {
```

Comment:
How about using the `read_x_into` variants of https://docs.rs/byteorder/1.3.2/byteorder/trait.ReadBytesExt.html instead of loops? This should be faster...

Reply:
Okay. For positions it's okay to loop for now though?

Reply:
Yeah, for positions and vectors of positions we need traits first, I think.